synthtool/transforms.py

```diff
@@ -133,7 +133,15 @@ def _copy_dir_to_existing_dir(
                 os.makedirs(str(dest_dir), exist_ok=True)
                 source_path = Path(os.path.join(root, name))
                 if merge is not None and dest_path.is_file():
-                    _merge_file(source_path, dest_path, merge)
+                    try:
+                        _merge_file(source_path, dest_path, merge)
+                    except Exception as e:
+                        logger.info(
+                            "_merge_file failed for %s: %s, fall back to copy",
+                            source_path,
+                            e,
+                        )
+                        shutil.copy2(str(source_path), str(dest_path))
```

Comment:
I'm surprised we want to fall back to copy rather than fail completely.  Have you seen a case where it's better to fall back to copy?

Reply:
PHP library has some binary files and synthtool dies when trying to `merge` those files.

Reply:
Ah, that's annoying.  To do this right would require running regex on the bytes.  But the signature for the _merge function requires strings, not bytes.  Ok.